optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/solution/descriptor/SolutionDescriptor.java

```diff
@@ -941,15 +988,17 @@ public boolean hasMovableEntities(ScoreDirector<Solution_> scoreDirector) {
      * @return {@code >= 0}
      */
     public long getGenuineVariableCount(Solution_ solution) {
-        return extractAllEntitiesStream(solution)
-                .mapToLong(entity -> findEntityDescriptorOrFail(entity.getClass()).getGenuineVariableCount())
-                .sum();
+        LongAdder adder = new LongAdder();
+        visitAllEntities(solution,
+                entity -> adder.add(findEntityDescriptorOrFail(entity.getClass()).getGenuineVariableCount()));
+        return adder.longValue();
     }
 
     public long getMaximumValueCount(Solution_ solution) {
-        return extractAllEntitiesStream(solution)
-                .mapToLong(entity -> findEntityDescriptorOrFail(entity.getClass()).getMaximumValueCount(solution, entity))
-                .max().orElse(0L);
+        LongAdder adder = new LongAdder();
+        visitAllEntities(solution,
+                entity -> adder.add(findEntityDescriptorOrFail(entity.getClass()).getMaximumValueCount(solution, entity)));
+        return adder.longValue();
```

Comment:
Copy-paste error. The method should return a `max()` of all entities' `maximumValueCount`, not their `sum()`. If you wait with this PR for Monday, you'll have this covered by a unit test.

Reply:
Well spotted. I will wait.